### PR TITLE
feat: get last 20 messages of a channel when joining the chat

### LINKED ISSUE #60: fetch previous chat history for a public channel

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #60

Makes it possible to get the last 20 messages from a channel when joining that channel.

Known issue: the messages will **not** be ordered. It needs a small adjustment to how we get new messages.

Just showing that all of this was on load: 
image

### CODE REVIEW

File: src/models/accounts.nim [hunk 1]
```diff
@@ -36,9 +37,15 @@ proc generateAddresses*(self: AccountModel): seq[GeneratedAccount] =
 proc generateRandomAccountAndLogin*(self: AccountModel) =
   let generatedAccounts = status_accounts.generateAddresses()
   let account = status_accounts.setupAccount(generatedAccounts[0], "qwerty")
+  # TODO this is needed for now for the retrieving of past messages. We'll either move or remove it later
+  let peer = "enode://44160e22e8b42bd32a06c1532165fa9e096eebedd7fa6d6e5f8bbef0440bc4a4591fe3651be68193a7ec029021cdb496cfe1d7f9f1dc69eb99226e6f39a7a5d4@35.225.221.245:443"
```
Comment (iurimatias) [inline src/models/accounts.nim:41]: this should be added to `setupAccount`

File: src/models/accounts.nim [hunk 2]
```diff
@@ -36,9 +37,15 @@ proc generateAddresses*(self: AccountModel): seq[GeneratedAccount] =
...
+  discard libstatus.addPeer(peer)
   self.events.emit("accountsReady", AccountArgs(account: account))
 
 proc storeAccountAndLogin*(self: AccountModel, selectedAccountIndex: int, password: string): Account =
   let generatedAccount: GeneratedAccount = self.generatedAddresses[selectedAccountIndex]
   result = status_accounts.setupAccount(generatedAccount, password)
+  # TODO this is needed for now for the retrieving of past messages. We'll either move or remove it later
+  let peer = "enode://44160e22e8b42bd32a06c1532165fa9e096eebedd7fa6d6e5f8bbef0440bc4a4591fe3651be68193a7ec029021cdb496cfe1d7f9f1dc69eb99226e6f39a7a5d4@35.225.221.245:443"
+  discard libstatus.addPeer(peer)
```
Comment (iurimatias) [inline src/models/accounts.nim:50]: same here

File: src/models/accounts.nim [hunk 3]
```diff
@@ -2,6 +2,7 @@ import eventemitter
 import json_serialization
 import ../status/accounts as status_accounts
 import ../status/types
+import ../status/libstatus
```
Comment (iurimatias) [inline src/models/accounts.nim:5]: importing libstatus outside of `status/` is forbidden :)

Comment (jrainville) [reply src/models/accounts.nim:5]: Yeah I know, but since it was only until we fix the TODO it could be accepted 🤷

**Pull request merged**